src/js/angular/core/services/repositories.service.js

```diff
@@ -187,7 +187,7 @@ repositories.service('$repositories', ['$http', 'toastr', '$rootScope', '$timeou
                 this.locationsShouldReload = false;
                 this.locations = [this.location];
                 const that = this;
-                return LocationsRestService.getLocations()
+                LocationsRestService.getLocations()
```

Comment:
If you don't return the promise here, how do you guarantee that a response has been received and locations are initialized before you actually ask for them? It's an asyncronous process. The service will sometimes return only `[local]` if there is some delay.

Try to add a non existing remote location (eg. http://non-existing). This will make the backend wait for a connection timeout. Go to create repository and refresh the page.